### PR TITLE
boost submodule

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: do_cmake.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ if test -e build; then
 fi
 mkdir build
 cd build
-cmake $@ ..
+cmake BOOST_J=$(nproc) $@ ..
```
Comment (cbodley) [inline do_cmake.sh]: i think the `BOOST_J=` should be `-DBOOST_J=`

**Comment by yuriw** [comment]
@liewegas I had problem running tests with this PR

"[yuriw@smithi018 build]$ git pull https://github.com/liewegas/ceph.git wip-denc
From https://github.com/liewegas/ceph
 * branch            wip-denc   -> FETCH_HEAD
Auto-merging src/os/bluestore/bluestore_types.h
CONFLICT (content): Merge conflict in src/os/bluestore/bluestore_types.h
Auto-merging src/os/bluestore/BlueStore.h
Auto-merging src/os/bluestore/BlueStore.cc
CONFLICT (content): Merge conflict in src/os/bluestore/BlueStore.cc
Auto-merging src/os/bluestore/BlueFS.cc
Automatic merge failed; fix conflicts and then commit the result."

**Comment by mattbenjamin** [comment]
@liewegas, @yuriw the first thing I've hit on centos7 is a conflict w/ceph manager boost check, fixing that, then moving on...

**Comment by mattbenjamin** [comment]
Wrt the actual build failure, it looks as if this CMake (2.8.10) CHECK_INCLUDE_FILES_CXX is behaving differently;  the proximate cause of failure is a failed check for boost/asio/coroutine.hpp, which does exist.

Also we need to add python to the list of boost libs being bootstrapped.  I'm inclined to just do that unconditionally.  I'll check actual build with -DCEPH_MGR=ON.

I think I'll need to create a branch with additional commits and mention it here.

**Comment by yuriw** [comment]
@mattbenjamin thanks, shall we remove `needs-qa` tag meanwhile?`

**Comment by liewegas** [comment]
https://github.com/ceph/ceph/pull/11817

**Pull request closed without merging**